src/components/ListItem.tsx

```diff
@@ -1,8 +1,70 @@
-import * as api from '../api';
-import './ListItem.css';
+import "./ListItem.css";
+import { updateItem, ListItem } from "../api";
+import { useEffect, useState } from "react";
+import toast from "react-hot-toast";
+import { is24HoursLater } from "../utils";
 
-type Props = Pick<api.ListItem, 'name'>;
+interface Props {
+	item: ListItem;
+	listPath: string | null;
+}
+
+export function ListItemCheckBox({ item, listPath }: Props) {
+	const [checked, setChecked] = useState(false);
+
+	const purchaseDate = item.dateLastPurchased
```

Comment:
Probably best to stick with the nullish coalescing operator here. It more explicitly communicates to the reader what you're handling here, and says more with less code 🙂  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing

Reply:
![purchaseDate](https://github.com/user-attachments/assets/96268648-1543-45b2-a45a-9e50b7df2758)


I was working on the change of using the nullish coalescing, and when I made the change I started getting a type error with it because in the function `!moreThan24HoursPassed(purchaseDate));` the type of Timestamp isn't Data. 
```
Argument of type 'Timestamp | Date' is not assignable to parameter of type 'Date'.
  Type 'Timestamp' is missing the following properties from type 'Date': toDateString, toTimeString, toLocaleDateString, toLocaleTimeString, and 37 more.ts(2345)
```

I don't necessarily think the 3rd option is the full go to either but I was curious if it would work. I guess it also could just be the `item.dateLastPurchased?.toDate() ?? null` but that feels kind of against the whole point of the nullish coalescing.

Reply:
Great insight @bbland1 and really helpful article @tannaurus !! I actually think the 3rd option is the best one for our specific use case! The case of item.dateLastPurchased being "null" or "undefined" is exactly what we want to capture so I think we can omit the null coalescing operator that would otherwise run the "right hand" evaluation if the "left-hand" side was null?!?  Thats my interpretation of the logic!  Really neat to see how this omission of the null coalescing operator is precisely because what we are clearly checking for ("null" or "undefined" purchase dates!)